File: cpp/1732-find-the-highest-altitude/find-the-highest-altitude.cpp
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int highest = 0;
        int altitude = 0;
        
        for (int n : gain) {
            altitude += n;
            highest = max(highest, altitude);
        }
        return highest;
    }
};

File: python/1732-find-the-highest-altitude/find-the-highest-altitude.py
class Solution:
    def largestAltitude(self, gain):
        """
        :type gain: list[int]
        :rtype: int
        """

        altitudes = [0]
        i = 0
        while i < len(gain):
            altitudes.append(altitudes[i]+gain[i])
            i+=1

        return max(altitudes)
